tests/ArmTemplates.Tests/Creator/TemplateCreatorTests/TagTemplateCreatorTests.cs

```diff
@@ -10,28 +10,150 @@
+            var creatorConfig = new CreatorParameters();
+
+            if (!tagNames.IsNullOrEmpty())
+            {
+                creatorConfig.Tags = new List<TagProperties>();
+                foreach (var tagName in tagNames)
+                {
+                    var tag = new TagProperties
+                    {
+                        DisplayName = tagName
+                    };
+                    creatorConfig.Tags.Add(tag);
+                }
+            }
+
+            if (!apiTagNames.IsNullOrEmpty())
+            {
+                creatorConfig.Apis = new List<ApiConfig>();
+                var apiTagNamesString = string.Join(", ", apiTagNames.ToArray());
+
+                var apiConfig = new ApiConfig
+                {
+                    Tags = apiTagNamesString
+                };
+
+                creatorConfig.Apis.Add(apiConfig);
+            }
+
+            return creatorConfig;
+        }
+
         [Fact]
-        public void ShouldCreateTagFromCreatorConfig()
+        public void CreateTagTemplate_ShouldCreateTemplateFromCreatorConfig_GivenApiTagsAndConfigTags()
         {
-            TagTemplateCreator tagTemplateCreator = new TagTemplateCreator(new TemplateBuilder());
-            CreatorParameters creatorConfig = new CreatorParameters() { Tags = new List<TagProperties>() };
-            TagProperties tag = new TagProperties()
+            var tagTemplateCreator = new TagTemplateCreator(new TemplateBuilder());
+            
+            var tagNames = new List<string>()
+            {
+                "tag 1", "tag2", "tag 3", "tag2"
+            };
+            var apiTagNames = new List<string>()
+            {
+                "tag 1", "tag2", "api tag 3", "api tag2"
+            };
+
+            var creatorConfig = this.GenerateCreatorParameters(tagNames, apiTagNames);
+
+            var expectedTagDictionary = new Dictionary<string, string>()
             {
-                DisplayName = "displayName"
+                { "tag-1", "tag 1" },
+                { "tag2", "tag2" },
+                { "tag-3" , "tag 3" },
+                { "api-tag-3", "api tag 3" },
+                { "api-tag2", "api tag2" }
             };
-            creatorConfig.Tags.Add(tag);
 
             //act
             var tagTemplate = tagTemplateCreator.CreateTagTemplate(creatorConfig);
-            var tagTemplateResource = (TagTemplateResource)tagTemplate.Resources[0];
 
             //assert
-            Assert.Equal($"[concat(parameters('{ParameterNames.ApimServiceName}'), '/{tag.DisplayName}')]", tagTemplateResource.Name);
+            tagTemplate.Parameters.Count().Should().Be(1);
+            tagTemplate.Parameters.ContainsKey(ParameterNames.ApimServiceName).Should().BeTrue();
+
+            tagTemplate.Resources.Count().Should().Be(5);
+            
+            foreach (TagTemplateResource tag in tagTemplate.Resources)
+            {
+                var resourceTagName = tag.Name.Split("/")[1].Split("'")[0];
```

Comment:
If I see this, I am all for saving original name of tag somewhere in resource. Such logic in tests is very error-prone IMO.

Reply:
Updated the method of checking with relying on common method GenerateParametrizedResourceName